// sinch-rest-api/papi/base_defs.js
module.exports = {
	getServerTime: {
		url: '/v1/timestamp/',
		method: 'GET',
		auth: 'nosign'
	},
	getInstance: {
		url: '/v1/instance',
		method: 'POST',
		auth: 'ticket'
	},
	renewInstance: {
		url: '/v1/instance',
		method: 'POST',
		auth: 'sign'
	},
	renewSecret: {
		url: '/v1/instances/id/:instanceId/authorization',
		method: 'PUT',
		auth: 'ticket'
	},
	//Experimental retrieval of instances (not working)
	getInstances: {
		url: '/v1/instances/id/',
		method: 'GET',
		auth: 'sign'
	},
	deleteInstance: {
		url: '/v1/instance/authorization',
		method:'DELETE',
		auth: 'sign'
	},
	getOrganisations: {
		url: '/v1/organisations',
		method: 'GET',
		auth: 'sign'
	},
	createOrganisation: {
		url: '/v1/organisations',
		method: 'POST',
		auth: 'sign'
	},
	updateOrganisation: {
		url: '/v1/organisations/id/:id',
		method: 'PATCH',
		auth: 'sign'
	},
	getUserSpace: {
		url: '/v1/organisations/id/:id/userspaces',
		method: 'GET',
		auth: 'sign'
	},
	getApplications: {
		url: '/v1/organisations/id/:id/applications',
		method: 'GET',
		auth: 'sign'
	},
	getApplication: {
		url: '/v1/applications/key/:key',
		method: 'GET',
		auth: 'sign'
	},
	createApplication: {
		url: '/v1/organisations/id/:id/applications',
		method: 'POST',
		auth: 'sign'
	},
	updateApplication: {
		url: '/v1/organisations/id/:id/applications/key/:key',
		method: 'PATCH',
		auth: 'sign'
	},
	getApplicationCredentials: {
		url: '/v1/applications/key/:key/credentials',
		method: 'GET',
		auth: 'sign'
	},
	getOrganisationAccounts: {
		url: '/v1/organisations/id/:orgId/accounts',
		method: 'GET',
		auth: 'sign'
	},
};
