// sinch-rest-api/papi/calling_defs.js
module.exports = {
	getConfiguration: {
		url: '/v1/configuration/user', 
		method: 'GET', 
		auth: 'sign'
	},
	placeCall: {
		url: '/v1/calls/:domain',
		method: 'POST',
		auth: 'sign'
	},
	callReporting: {
		url: '/v1/calls/:domain/id/:callId',
		method: 'PUT',
		auth: 'sign'
	},
	postMedia: {
		url: '/v1/media',
		method: 'POST',
		auth: 'sign'
	},
	queryNumber: {
		url: '/v1/calling/query/number/:number',
		method: 'GET',
		auth: 'sign'
	},
	manageCall: {
		url: '/v1/calls/id/:callId',
		method: 'PATCH',
		auth: 'sign'
	},
	manageConference: {
		url: '/v1/conferences/id/:conferenceId/:callId',
		method: 'PATCH',
		auth: 'sign'
	},
	deleteConferenceParticipant: {
		url: '/v1/conferences/id/:conferenceId/:callId',
		method: 'DELETE',
		auth: 'sign'
	},
	deleteConference: {
		url: '/v1/conferences/id/{conferenceId}',
		method: 'DELETE',
		auth: 'sign'
	},
	callouts: {
		url: '/v1/callouts',
		method: 'POST',
		auth: 'sign'
	},
	getCallResult: {
		url: '/v1/calls/id/:callId',
		method: 'GET',
		auth: 'sign'
	}
};
