// sinch-rest-api/papi/portal_defs.js
module.exports = {
	getRequestForgotPasswordEmail: {
		url: '/v1/users/email/:email/password/forgot',
		method: 'POST',
		auth: 'sign'
	},
	sendWelcomeEmail: {
		url: '/v1/messages',
		method: 'POST',
		auth: 'sign'
	},
	postOrder: {
		url: '/v1/organisations/id/:orgId/transactions',
		method: 'POST',
		auth: 'sign'
	},
	submitOrder: {
		url: '/v1/transactions/id/:id/submit',
		method: 'POST',
		auth: 'sign'
	},
	reportPayPalTransaction: {
		url: '/v1/transactions/id/:transactionId/submit',
		method: 'PUT',
		auth: 'sign'
	},
	paymentProfile: {
		url: '/v1/organisations/id/:id/paymentprofile',
		method: 'GET',
		auth: 'sign'
	},
	createAutoTopup: {
		url: '/v1/organisations/id/:id/paymentprofile/autotopups',
		method: 'POST',
		auth: 'sign'
	},
	modifyAutoTopup: {
		url: '/v1/organisations/id/:id/paymentprofile/autotopups/:autoTopUpId',
		method: 'PUT',
		auth: 'sign'
	},
	deletePaymentInfo: {
		url: '/v1/organisations/id/:id/paymentprofile/paymentinfos/:paymentinfoId',
		method: 'DELETE',
		auth: 'sign'
	},
	pollOrder: {
		url: '/v1/transactions/id/:transactionId',
		method: 'GET',
		auth: 'sign'
	},
	su: {
		url: '/v1/poweruser/email/:email/authentication',
		method: 'GET',
		auth: 'sign'
	},
	validateVat: {
		url: '/v1/vatnumber/:vatnumber',
		method: 'GET',
		auth: 'sign'
	},
	logSale: {
		url: '/v1/sales/',
		method: 'POST',
		auth: 'sign'
	},
	// DID Management
	getNumberPools: {
		url: '/v1/organisations/id/:id/numbers/shop',
		method: 'GET',
		auth: 'sign'
	},
	reserveFromPool: {
		url: '/v1/organisations/id/:id/numbers/shop',
		method: 'PUT',
		auth: 'sign'
	},
	checkoutReservations: {
		url: '/v1/organisations/id/:id/numbers/shop',
		method: 'POST',
		auth: 'sign'
	},
	cancelAllReservations: {
		url: '/v1/organisations/id/:id/numbers/shop',
		method: 'DELETE',
		auth: 'sign'
	},
	getMyNumbers: {
		url: '/v1/organisations/id/:id/numbers',
		method: 'GET',
		auth: 'sign'
	},
	getNumbersForApp: {
		url: '/v1/applications/key/:key/numbers',
		method: 'GET',
		auth: 'sign'
	},
	setNumbersForApp: {
		url: '/v1/applications/key/:applicationKey/numbers/:numberCapability',
		method: 'PUT',
		auth: 'sign'
	},
};
