// sinch-rest-api/papi/reporting_defs.js
module.exports = {
	getReportCalls: {
		url: '/v1/reports/group/calls/data',
		method: 'GET',
		auth: 'sign'
	},
	getReportPSTN: {
		url: '/v1/reports/group/calls/pstn',
		method: 'GET',
		auth: 'sign'
	},
	getReportMessages: {
		url: '/v1/reports/group/messages/data',
		method: 'GET',
		auth: 'sign'
	},
	getReportUsers: {
		url: '/v1/reports/group/activeusers/:period',
		method: 'GET',
		auth: 'sign'
	},
	getReportSMS: {
		url: '/v1/reports/group/messages/sms',
		method: 'GET',
		auth: 'sign'
	},
	getReportVerifications: {
		url: '/v1/reports/verifications',
		method: 'GET',
		auth: 'sign'
	},
	getCDRReportPSTN: {
		url: '/v1/reports/cdr/pstn',
		method: 'GET',
		auth: 'sign'
	},
	getCDRReportSMS: {
		url: '/v1/reports/cdr/sms',
		method: 'GET',
		auth: 'sign'
	},
	getCDRReportData: {
		url: '/v1/reports/cdr/data',
		method: 'GET',
		auth: 'sign'
	},
	getCDRPstnIn: {
		url: '/v1/reports/cdr/pstnorg',
		method: 'GET',
		auth: 'sign'
	},
	getCDRVerification: {
		url: '/v1/reports/cdr/verification',
		method: 'GET',
		auth: 'sign'
	}
};
